### PR TITLE
test: Adjust tests to use new `runServerless` syntax

### PR DESCRIPTION
**Pull request by pgrzesik**

### CODE REVIEW

File: test/unit/lib/plugins/aws/provider.test.js [hunk 1]
```diff
@@ -474,30 +474,12 @@ role_arn = NOTDEFAULTWITHROLEROLE
       expect(awsCredentials.credentials.accessKeyId).to.equal('NOTDEFAULTKEYID');
     });
 
-    it('should get credentials from `provider.credentials`', async () => {
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/provider.test.js:477]: Why do we remove that test?

Comment (pgrzesik) [reply test/unit/lib/plugins/aws/provider.test.js:477]: It should never be there in the first place as `provider.credentials` is not supported by schema. We can still introduce it to schema I guess, but we also have this issue that will fully deprecate and remove it: https://github.com/serverless/serverless/issues/8356

What do you think?

Comment (medikoo) [reply test/unit/lib/plugins/aws/provider.test.js:477]: I think we can have that test removed. It should also be deprecated in context of v2

**Review by medikoo — APPROVED** [review]
Ok, let's have thius

**Pull request merged**